### ISSUE TITLE
`reportUnsafeMultipleInheritance` false positive when diamond

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#657]
Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&reportUnusedParameter=false&code=MYGwhgzhAEAqD2AHAXAWAFDS9AJgUwDNoB9YgSwDsyAXUgCgjxAIEploA6LjH9USGABlC1OgkRtO3dL35RoAJTIBzABajxkrh14Y5MACLwA7hTrCC1ADSKV6rVywBiLHgBOb%2BG-YBZAK4g1GSIIHjQlKruNGAUwGFkMBTw1NBgICAmeDjQAEZ4wGB%2BjNDUkdAE8OkmlMq5kGH6eDDA8BTUYJTQAAaklDSkXdBe3aQUeMYD0AC2eKXwODClYClTYACe0EkpyrPQBelZ7ACC0HRueIhe1ACqFBBgBHj%2BgcGhAJIUkW7RsXgsvIhvm06EZTBwpp46CwWM5oKCKDYLNZbGpkeIgA)

```python
class Up:
    def __init__(self): ...


class Left(Up): ...


class Right(Up): ...


class Down(Left, Right): ...  #  error: Multiple inheritance is not allowed because the following base classes contain `__init__` or `__new__` methods that may not get called: Up (reportUnsafeMultipleInheritance)

print(Down.mro())  # Down, Left, Right, Up
```
